Tests/MongoSwiftTests/UnifiedTestRunner/UnifiedTestRunner.swift

```diff
@@ -248,17 +254,52 @@ struct UnifiedTestRunner {
                             }
                         }
                     }
+                    // Manual clean up
+                    try await self.terminateOpenTransactions()
+                    await context.disableFailpoints()
+                    try await self.closeEntities(context: context)
                 } catch let testErr {
                     // Test runners SHOULD terminate all open transactions after each failed test by killing all
                     // sessions in the cluster.
                     do {
-                        try self.terminateOpenTransactions()
+                        // Clean up to prevent a client closing error from obscuring the real error
+                        try? await self.terminateOpenTransactions()
+                        await context.disableFailpoints()
+                        try await closeEntities(context: context)
                     } catch {
                         print("Failed to terminate open transactions: \(error)")
                     }
                     throw testErr
                 }
             }
         }
+
+        try self.internalClient.close()
+    }
+
+    func closeEntities(context: Context) async throws {
+        // Must close streams and cursors before closing the parent client
+        for entity in context.entities.values {
```

Comment:
we actually probably should close any cursor entities here too, I think. it might just happen that all tests right now iterate cursors to completion (which leads to them being closed) but that might not always be the case for future tests.